### ISSUE TITLE
[Crackle] 'NoneType' object has no attribute 'attrib'

### ISSUE THREAD

**Issue by stepshal** [rg3/youtube-dl#10333]
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2016.08.12**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

```
$ y -v 'http://www.crackle.com/the-art-of-more/2496419'
[debug] System config: []
[debug] User config: ['-4', '-f', 'best', '--prefer-free-formats', '--no-cache-dir', '--no-mtime', '--youtube-skip-dash-manifest']
[debug] Command-line args: ['-v', 'http://www.crackle.com/the-art-of-more/2496419']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.08.12
[debug] Python version 3.5.2+ - Linux-4.7.0-rc7-686-i686-with-debian-stretch-sid
[debug] exe versions: ffmpeg 3.1.2, ffprobe 3.1.2, rtmpdump 2.4
[debug] Proxy map: {}
[Crackle] 2496419: Downloading XML
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl", line 9, in <module>
    load_entry_point('youtube-dl==2016.8.12', 'console_scripts', 'youtube-dl')()
  File "/usr/local/lib/python3.5/dist-packages/youtube_dl/__init__.py", line 433, in main
    _real_main(argv)
  File "/usr/local/lib/python3.5/dist-packages/youtube_dl/__init__.py", line 423, in _real_main
    retcode = ydl.download(all_urls)
  File "/usr/local/lib/python3.5/dist-packages/youtube_dl/YoutubeDL.py", line 1786, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "/usr/local/lib/python3.5/dist-packages/youtube_dl/YoutubeDL.py", line 691, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/lib/python3.5/dist-packages/youtube_dl/extractor/common.py", line 347, in extract
    return self._real_extract(url)
  File "/usr/local/lib/python3.5/dist-packages/youtube_dl/extractor/crackle.py", line 54, in _real_extract
    title = item.attrib['t']
AttributeError: 'NoneType' object has no attribute 'attrib'
```
http://www.crackle.com/the-art-of-more/2496419

**Comment by yan12125** [comment]
In Chrome this video said: "Uh-oh, something went wrong.". Do you have other examples?

**Comment by stepshal** [comment]
@yan12125 http://www.crackle.com/comedians-in-cars-getting-coffee/2498934
Replace this with broken url then, please.

**Comment by remitamine** [comment]
geo restricted, works using US ip:
```
youtube-dl -F http://www.crackle.com/the-art-of-more/2496419
[Crackle] 2496419: Downloading XML
[Crackle] 2496419: Downloading m3u8 information
[info] Available formats for 2496419:
format code      extension  resolution note
hls-meta         mp4        multiple   Quality selection URL 
hls-127          mp4        192x108     127k , avc1.42000b, 15.0fps, mp4a.40.5
hls-253          mp4        256x144     253k , avc1.42000c, 30.0fps, mp4a.40.5
hls-475          mp4        416x234     475k , avc1.42000d, 30.0fps, mp4a.40.5
http-c544        flv        544x306    
http-360p        mp4        640x360    
hls-935          mp4        704x396     935k , avc1.4d001e, 30.0fps, mp4a.40.5
http-480p        mp4        852x478    
http-480p_1mbps  mp4        852x478    
hls-1827         mp4        896x504    1827k , avc1.4d001f, 30.0fps, mp4a.40.5
hls-3837         mp4        1280x720   3837k , avc1.4d001f, 30.0fps, mp4a.40.5 (best)
```

**Comment by yan12125** [comment]
PS. The M3U8 manifest is geo-restricted, while HTTP streams not.

**Comment by remitamine** [comment]
the m3u8 manifest s also not geo restricted.

**Comment by yan12125** [comment]
Hmm, I got this in Taiwan:
```
$ youtube-dl -vF "http://www.crackle.com/comedians-in-cars-getting-coffee/2498934"
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-vF', 'http://www.crackle.com/comedians-in-cars-getting-coffee/2498934']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.08.13
[debug] Git HEAD: 6d8ec8c
[debug] Python version 3.5.2 - Linux-4.7.0-1-ARCH-x86_64-with-arch
[debug] exe versions: ffmpeg 3.1.2, ffprobe 3.1.2, rtmpdump 2.4
[debug] Proxy map: {}
[Crackle] 2498934: Downloading XML
[Crackle] 2498934: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 404: Not Found
[info] Available formats for 2498934:
format code      extension  resolution note
http-c544        flv        544x306    
http-360p        mp4        640x360    
http-480p        mp4        852x478    
http-480p_1mbps  mp4        852x478    (best)
```

**Comment by remitamine** [comment]
can you access uplynk m3u8 manifests from other sites(abc7news, viceland ...).

**Comment by yan12125** [comment]
Only the Crackle example fails: (I've added some codes to display the M3U8 URL)
```
$ youtube-dl -vF test:viceland test:abc7news "http://www.crackle.com/comedians-in-cars-getting-coffee/2498934"
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-vF', 'test:viceland', 'test:abc7news', 'http://www.crackle.com/comedians-in-cars-getting-coffee/2498934']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.08.13
[debug] Git HEAD: 6d8ec8c
[debug] Python version 3.5.2 - Linux-4.7.0-1-ARCH-x86_64-with-arch
[debug] exe versions: ffmpeg 3.1.2, ffprobe 3.1.2, rtmpdump 2.4
[debug] Proxy map: {}
[TestURL] Test URL: https://www.viceland.com/en_us/video/cyberwar-trailer/57608447973ee7705f6fbd4e
[Viceland] 57608447973ee7705f6fbd4e: Downloading webpage
[Viceland] 57608447973ee7705f6fbd4e: Downloading JSON metadata
[uplynk:preplay] 59c285c5471d4273a0d5f70642949b51: Downloading JSON metadata
http://content.uplynk.com/59c285c5471d4273a0d5f70642949b51.m3u8
[uplynk:preplay] 59c285c5471d4273a0d5f70642949b51: Downloading m3u8 information
[uplynk:preplay] 59c285c5471d4273a0d5f70642949b51: Downloading JSON metadata
[info] Available formats for 57608447973ee7705f6fbd4e:
format code  extension  resolution note
meta         mp4        multiple   Quality selection URL 
127          mp4        192x108     127k , avc1.42000b, 15.0fps, mp4a.40.5
229          mp4        256x144     229k , avc1.42000c, 15.0fps, mp4a.40.5
English      mp4        416x234     427k , avc1.42000d, 30.0fps, mp4a.40.5
733          mp4        704x396     733k , avc1.4d001e, 30.0fps, mp4a.40.5
1261         mp4        896x504    1261k , avc1.4d001f, 30.0fps, mp4a.40.5
2653         mp4        1280x720   2653k , avc1.4d001f, 30.0fps, mp4a.40.5
4423         mp4        1920x1080  4423k , avc1.640028, 30.0fps, mp4a.40.5 (best)
[TestURL] Test URL: http://abc7news.com/entertainment/east-bay-museum-celebrates-vintage-synthesizers/472581/
[Abc7News] east-bay-museum-celebrates-vintage-synthesizers: Downloading webpage
http://content.uplynk.com/ext/4413701bf5a1488db55b767f8ae9d4fa/011215-kgo-4pm-museum-vid_web.m3u8?ad._v=2&ad=kgo_video&ad.preroll=&ad.fill_slate=1&ad.ametr=1
[Abc7News] east-bay-museum-celebrates-vintage-synthesizers: Downloading m3u8 information
[info] Available formats for 472581:
format code  extension  resolution note
meta         mp4        multiple   Quality selection URL 
183          mp4        192x108     183k , avc1.42000b, 15.0fps, mp4a.40.5
354          mp4        256x144     354k , avc1.42000c, 30.0fps, mp4a.40.5
628          mp4        416x234     628k , avc1.42000d, 30.0fps, mp4a.40.5
1070         mp4        704x396    1070k , avc1.4d001e, 30.0fps, mp4a.40.5
1859         mp4        896x504    1859k , avc1.4d001f, 30.0fps, mp4a.40.5
3872         mp4        1280x720   3872k , avc1.4d001f, 30.0fps, mp4a.40.5 (best)
[Crackle] 2498934: Downloading XML
http://content.uplynk.com/ext/e8773f7770a44dbd886eee4fca16a66b/2498934.m3u8
[Crackle] 2498934: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 404: Not Found
[info] Available formats for 2498934:
format code      extension  resolution note
http-c544        flv        544x306    
http-360p        mp4        640x360    
http-480p        mp4        852x478    
http-480p_1mbps  mp4        852x478    (best)
```
The problematic m3u8 returns 404 even with an US proxy.
```
$ curl -v http://content.uplynk.com/ext/e8773f7770a44dbd886eee4fca16a66b/2498934.m3u8 --proxy http://127.0.0.1:8087/
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 8087 (#0)
> GET http://content.uplynk.com/ext/e8773f7770a44dbd886eee4fca16a66b/2498934.m3u8 HTTP/1.1
> Host: content.uplynk.com
> User-Agent: curl/7.50.1
> Accept: */*
> 
< HTTP/1.1 404 OK
< Content-Length: 36
< X-Services: 6d38de1813c542a4ad876dfb5c128460
< Server: uplynk webStack/2.4
< Date: Mon, 15 Aug 2016 11:48:28 GMT
< Access-Control-Allow-Origin: *
< X-Human: Hello, fellow human! You should come work with us! uljobs@verizondigitalmedia.com. Mention this header.
< Content-Type: text/html
< 
<html><body>Not found</body></html>
* Connection #0 to host 127.0.0.1 left intact
```

**Comment by remitamine** [comment]
i think there is something wrong here, i start getting 404 error now.
so i manifest doesn't fail because of geo restriction, there something that makes them think that the request are not from the browser.

**Comment by remitamine** [comment]
i found the error, for this url http://www.crackle.com/comedians-in-cars-getting-coffee/2498934 the `owner_id` is diffrent, it should be `6790c94a453741799d3f3fb3bc9023a4`.
so this url http://content.uplynk.com/ext/6790c94a453741799d3f3fb3bc9023a4/2498934.json should work.

**Comment by yan12125** [comment]
Thanks for the report. Videos on Crackle.com will work again in the next version.

**Comment by anisse** [comment]
@yan12125 it seems b54a2da4333556baa3b34fc595060223181320d1 introduced a regression, see:
https://travis-ci.org/anisse/youtube-dl/jobs/155080351

**Comment by yan12125** [comment]
Thanks @anisse it's indeed a regression, which should be fixed in 906b87cf5f6ccf28ebd75d6a92367d7c238f2ad9
